### DISCUSSION TITLE
Changeable Cursor

### DISCUSSION THREAD

**Discussion by zyberzebra** [varabyte/kotter#123] [Q&A]
**Is your feature request related to a problem? Please describe.**
 I would like to be able to change the cursor style and shape in Kotter. Currently, the cursor is always the default style and shape.

**Describe the solution you'd like**
 I would like to be able to change the cursor style and shape in Kotter, or maybe even hiding the cursor.

**Describe alternatives you've considered**
none

**Additional context**
![image](https://github.com/user-attachments/assets/f34319e3-2a34-43b4-8869-83cf31ab3d58)

**Comment by bitspittle** [comment]
Hello, and very sorry for not replying until now.

Can you share code that reproduces this? Are you using an `input()` somewhere? Otherwise, I'm not seeing a cursor on my machine.

**Comment by zyberzebra** [comment]
Sure! https://github.com/zyberzebra/Terminal-Quiz

https://github.com/zyberzebra/Terminal-Quiz/blob/master/src/main/kotlin/Main.kt

**Comment by bitspittle** [comment]
Why are you using `input` there? Just handle enter pressed and choose the active line?

**Comment by zyberzebra** [comment]
Oh… I see. I'm gonna try that tomorrow.

Sorry for bothering and thanks for helping!

**Reply by bitspittle** [reply]
By the way this wasn't a bother. When you ask questions in public, it helps other people. 

I hope you were able to learn from the `list` example and make progress!

**Comment by bitspittle** [comment]
So just to make sure you understand the point of `input`, it exists purely for freeform text to be typed in. I assume you've gone through the README already but just in case [here is the relevant section](https://github.com/varabyte/kotter#%EF%B8%8F-user-input)

You would almost certainly be interested in the [`list` code example](https://github.com/varabyte/kotter/blob/main/examples/list/src/main/kotlin/main.kt):

```
$ git clone https://github.com/varabyte/kotter
$ cd kotter/examples/list
$ ../../gradlew run
```
